src/components/App.js

```diff
@@ -14,30 +14,110 @@ function App() {
   /* const [inputFields, setInputFields] = useState(data);
   console.log(inputFields); */
 
+
   useEffect(() => {
     axios.get('http://localhost:5000/resumes/5fbee248eb620e488cfc11f8')
       .then(response => setData(response.data))
     setLoading(false);
+
   }, [])
 
-  
+  const toggleView = () => setEditMode(!isEditMode);
+
+  /* const handleInfoChange = useCallback(
+    (newData) => {
+      setData(
+        {
+          ...data,
+          firstName: newData?.newfirstName,
+          lastName: newData?.newlastName,
+          position: newData?.newposition,
+          email: newData?.newemail
+        });
+    },
+    [data]
+  ); */
+
+
   const handleInfoChange = (newData) => {
-   setData(
-    {
-    ...data,
-    firstName: newData?.newfirstName,
-    lastName: newData?.newlastName,
-    position: newData?.newposition,
-    email: newData?.newemail
-   })
-   console.log("TRIGER IN APP: ", newData)
+    setData(
+      {
+        ...data,
+        firstName: newData?.newfirstName,
+        lastName: newData?.newlastName,
+        position: newData?.newposition,
+        email: newData?.newemail
+      })
+
+
+  }
+
+  const handleQuoteChange = (newData) => {
+    setData(
+      {
+        ...data,
+        quote: newData?.newquote,
+
+      })
+
+
+  }
+
+  const handleBiographyChange = (newData) => {
+    setData(
+      {
+        ...data,
+        biography: newData?.newbiography,
+
+      })
+
+
+  }
+
+  const handlePictureChange = (newData) => {
+    setData(
+      {
+        ...data,
+        picture: newData?.newpicture,
+
+      })
+
+
+  }
+
+  const handlePersonalInfoChange = (newData) => {
+    setData(
+      {
+        ...data,
+        dob: newData?.newdob,
+        pob: newData?.newpob,
+      })
+
+
+  }
+
+  const handleExperiencesChange = (newData) => {
+    setData(
+      {
+        ...data,
+        duration: newData?.newdob,
+
+      })
```

Comment:
```suggestion
   const handleChange = (e) => {
     const { target: { name, value }} = e;
     setData({
     ...data,
     [name]: value
     })
   } 
```

and try passing this function everywhere that you need to change an input field, it shoud work in theory. You were close with your solution.. there is no need to do extra stuff like the combination of useEffect and useState you made inside of every component. If you have questions about this we can talk either here directly or on discord :D